### DISCUSSION TITLE
How can I use the cookie list?

### DISCUSSION THREAD

**Discussion by ghost** [MicrosoftEdge/WebView2Feedback#3867] [Q&A] [answered]
I have List<string> cookie = new List<string>(); <---- cookie list 14 string NetScape

 How can I use the cookie list?
 
 https://social.msdn.microsoft.com/Forums/en-US/1983408d-f9d9-44fd-a0c3-88b30d757644/how-to-set-cookies-to-webview2-control-in-c?forum=csharpgeneral
 
 This method did not help me

**Comment by ghost** [comment]
Winforms Webview2 ver. - 1.0.774.44

**Comment by champnic** [comment]
Hey @ZalupaSlonika1488 - Can you add some context on what you are trying to do here? What did you try and what problems did you run into?

**Comment by ghost** [comment]
> Привет, @ ZalupaSlonika1488 - Не могли бы вы добавить контекст того, что вы пытаетесь здесь сделать? Что вы пробовали и с какими проблемами столкнулись?

Hi. I wanted to log in to the site using cookies. 

I tried different methods, but I never managed to send 14 cookies to the site. Maybe you know how to do it? Either via Cookies or via AddHeaders.

I will be extremely grateful for any help.

Unfortunately, I don't have any code left.

**Comment by champnic** [comment, marked as answer]
Sure thing!

You should be able to get the list of cookies by using [WebView2.CoreWebView2.CookieManager](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.cookiemanager?view=webview2-dotnet-1.0.774.44#Microsoft_Web_WebView2_Core_CoreWebView2_CookieManager), and then adding the cookies you want using [CookieManager.AddOrUpdateCookie(...)](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2cookiemanager.addorupdatecookie?view=webview2-dotnet-1.0.774.44#Microsoft_Web_WebView2_Core_CoreWebView2CookieManager_AddOrUpdateCookie_Microsoft_Web_WebView2_Core_CoreWebView2Cookie_). Can you give that a try?
